lib/msal-browser/src/controllers/NestedAppAuthController.ts

```diff
@@ -0,0 +1,396 @@
+    }
+
+    private async acquireTokenInteractive(
+        request: PopupRequest | RedirectRequest
+    ): Promise<AuthenticationResult> {
+        const naaRequest = this.nestedAppAuthAdapter.toNaaTokenRequest(request);
+        const response = await this.bridgeProxy.getTokenInteractive(naaRequest);
+
+        return this.nestedAppAuthAdapter.fromNaaTokenResponse(
+            naaRequest,
+            response
+        );
+    }
+
+    private async acquireTokenSilentInternal(
+        request: SilentRequest
+    ): Promise<AuthenticationResult> {
+        const naaRequest =
+            this.nestedAppAuthAdapter.toNaaSilentTokenRequest(request);
+        const response = await this.bridgeProxy.getTokenSilent(naaRequest);
+
+        return this.nestedAppAuthAdapter.fromNaaTokenResponse(
+            naaRequest,
+            response
+        );
+    }
+
+    async acquireTokenPopup(
+        request: PopupRequest
+    ): Promise<AuthenticationResult> {
+        return this.acquireTokenInteractive(request);
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    acquireTokenRedirect(request: RedirectRequest): Promise<void> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+
+    async acquireTokenSilent(
+        silentRequest: SilentRequest
+    ): Promise<AuthenticationResult> {
+        return this.acquireTokenSilentInternal(silentRequest);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    acquireTokenByCode(
+        request: AuthorizationCodeRequest // eslint-disable-line @typescript-eslint/no-unused-vars
+    ): Promise<AuthenticationResult> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    acquireTokenNative(
+        request: // eslint-disable-line @typescript-eslint/no-unused-vars
+        | SilentRequest
+            | Partial<
+                  Omit<
+                      CommonAuthorizationUrlRequest,
+                      | "requestedClaimsHash"
+                      | "responseMode"
+                      | "codeChallenge"
+                      | "codeChallengeMethod"
+                      | "nativeBroker"
+                  >
+              >
+            | PopupRequest,
+        apiId: ApiId, // eslint-disable-line @typescript-eslint/no-unused-vars
+        accountId?: string | undefined // eslint-disable-line @typescript-eslint/no-unused-vars
+    ): Promise<AuthenticationResult> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    acquireTokenByRefreshToken(
+        commonRequest: CommonSilentFlowRequest, // eslint-disable-line @typescript-eslint/no-unused-vars
+        silentRequest: SilentRequest // eslint-disable-line @typescript-eslint/no-unused-vars
+    ): Promise<AuthenticationResult> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    addEventCallback(callback: Function): string | null {
```

Comment:
@tnorling IIRC these are necessary to support Angular/React -- what do these frameworks except us to do here?

Reply:
There's code in each of the acquireToken functions which emits "events" (such as Login started, succeeded, failed, etc). Essentially it's invoking any callbacks which have been registered using this API. You can look at the standard implementation to make more sense of it.